imod_coupler/__main__.py

```diff
@@ -94,53 +103,53 @@ def main():
         version="%(prog)s {version}".format(version=__version__),
     )
 
+    if len(sys.argv) == 1:
+        sys.argv.append("--help")
     args = parser.parse_args()
 
     logging.basicConfig(level=args.log_level)
     mf6_dll = args.mf6_dll
     msw_dll = args.msw_dll
     mf6_model_dir = args.mf6_model_dir
     msw_model_dir = args.msw_model_dir
+    configfile = args.config
     msw_mpi_dll_dir = args.msw_mpi_dll_dir
     debug_native = args.enable_debug_native
     timing = args.timing
 
     if timing:
         start = time.perf_counter()
 
-    if not os.path.exists(mf6_dll):
-        logger.error(f"MODFLOW6 dll {mf6_dll} not found.")
-        sys.exit(1)
-
-    if not os.path.exists(msw_dll):
-        logger.error(f"METASWAP dd {msw_dll} not found.")
-        sys.exit(1)
-
-    if not os.path.isdir(mf6_model_dir):
-        logger.error(f"MODFLOW6 Model path {mf6_model_dir} not found.")
-        sys.exit(1)
-
-    if not os.path.isdir(msw_model_dir):
-        logger.error(f"MetaSWAP Model path {msw_model_dir} not found.")
-        sys.exit(1)
-
-    if not os.path.isdir(msw_mpi_dll_dir):
-        logger.error(f"Metaswap MPI dlls {msw_mpi_dll_dir} not found.")
-        sys.exit(1)
-
     # wait for native debugging
     if debug_native:
         input(f"PID: {os.getpid()}, press any key to continue ....")
 
+    config_data = \
+        {'components': {}, 'dependencies': [], 'parameters': [], 'exchanges': []}
```

Comment:
I think we want to have a Config class, instead of a dict here.
For inspiration you can see https://github.com/anoadragon453/nio-template/blob/master/my_project_name/config.py

Reply:
I also think that the config logic should not be done in __main__ but rather in metamod, so that it can be used as a library. So e.g. metamod should receive a path to the config file and then does the rest itself.